kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/AbstractKieServicesClientImpl.java

```diff
@@ -861,7 +862,11 @@ protected KieServerHttpRequest invoke(String url, RemoteHttpOperation operation)
                 if (e.getCause() instanceof IOException) {
                     logger.warn("Marking endpoint '{}' as failed due to {}", url, e.getCause().getMessage());
                     String failedBaseUrl = loadBalancer.markAsFailed(url);
-                    nextUrl = loadBalancer.getUrl();
+                    try {
+                        nextUrl = loadBalancer.getUrl();
+                    } catch (NoEndpointFoundException noEndpointFoundException) {
```

Comment:
What should happen in case no endpoint is found?
According to your PR the execution flow will still run in while loop. AFAIK this is not a correct approach.

Reply:
when load balancer does not has an endpoint, nextUrl will be null and break the loop.
 it will throw KieServerHttpRequestException.

Reply:
if it does not have an endpoint, nextUrl will be null, it will break the loop[1].
The code will throw KieServerHttpRequestException [2].The KieServerHttpRequestException will catch by UI.

[1] https://github.com/kiegroup/droolsjbpm-integration/pull/2422/files/dfaf0d15e2efad5def94eb6512402a5d507253ad#diff-2e9dbb035c9f9deb6c6e3a76a2480f754db0f6221a0d5f8080eb8436b58968f4R876
[2] https://github.com/kiegroup/droolsjbpm-integration/pull/2422/files/dfaf0d15e2efad5def94eb6512402a5d507253ad#diff-2e9dbb035c9f9deb6c6e3a76a2480f754db0f6221a0d5f8080eb8436b58968f4R878

Reply:
Right, I overlooked the condition in the `while`, thanks